### PR TITLE
deploy netdata using the nodiscc.xsrv.monitoring_netdata ansible role

### PR DESCRIPTION
**Pull request by nodiscc**
- fixes https://gitlab.com/nodiscc/xsrv/-/issues/1
- README.md: fix typo, whitespace

/cc @odyslam

### CODE REVIEW

**Comment by nodiscc** [comment]
/cc @ilyam8 @cakrit it seems you are the new maintainers.

Are you still interested in adding this documentation?

File: netdata-agent-deployment/ansible-xsrv/README.md
Comment (ilyam8) [inline netdata-agent-deployment/ansible-xsrv/README.md]: Screenshot 2022-05-04 at 20 56 47

Comment (nodiscc) [reply netdata-agent-deployment/ansible-xsrv/README.md]: https://asciinema.org/a/bpWXRpyOrBQyJHaFusbM8Slkb:

> This recording has been archived
> All unclaimed recordings (the ones not linked to any user account) are automatically archived 7 days after upload.

I did not know that, I will probably redo it when I get some time (I did not keep the source file), I have removed the screencast section for the time being.

**Comment by nodiscc** [comment]
@ilyam8 ok to merge?

**Comment by ilyam8** [comment]
@nodiscc sorry for the delay. @Ferroin and @maneamarius will take a look (I hope) soon.

**Comment by ReifiedException** [comment]
As far as I remember, it was decided that we will not offer the community automation solutions. We had a meeting. Why we are talking about it again?

**Comment by andrewm4894** [comment]
If the pr might help some future users why would we not merge it?

Also the pr seems to add a new tree to the folder so it's not like it's going to break anything. 

Without being an expert or even all that knowledgeable on Ansible this looks like a great community contribution to me and exactly the sort of thing this repo is for.

**Comment by ReifiedException** [comment]
Going back to that discussion, the reasons for not doing it are roughly the following.
This way we sort of take responsibility for this code. We would be more comfortable giving links to the repositories of those who developed the relevant playbook/role.
This particular role is (from memory) well written, but its usefulness is extremely limited by the fact that it only works with
Debian 11. Since there are no organizations that use a single version of a single operating system, more generic roles will be preferred.
Also, Ansible is not the only automation tool.
I would prefer to see this role in Ansible Galaxy. And we will, of course, point out its existence in the documentation.

**Review by Ferroin — APPROVED** [review]

**Review by maneamarius — APPROVED** [review]

**Comment by andrewm4894** [comment]
Thanks @nodiscc !

**Comment by nodiscc** [comment]
Hi @andrewm4894 @iigorkarpov 

> This way we sort of take responsibility for this code. We would be more comfortable giving links to the repositories of those who developed the relevant playbook/role.

I assumed (see https://gitlab.com/nodiscc/xsrv/-/issues/1) this repository was for listing community-supported projects, which implies netdata does not officially support/take responsibility for the content.

If you feel more comfortable *only* linking to external projects I suggest converting the repo to an "awesome"-style list (see https://gitea.com/gitea/awesome-gitea for example) which links to community projects or articles on netdata. I can move the article to a Github gist or similar then we can simply link to it from here (then we should also move the other ansible howto).

All things considered I think the "awesome list" format would be more suitable for this repo.

> This particular role is (from memory) well written,

Thanks :ok_hand: I try to keep it clean

> its usefulness is extremely limited by the fact that it only works with Debian 11. Since there are no organizations that use a single version of a single operating system, more generic roles will be preferred.

It should work with Debian 10/Ubuntu 18/20/22 with minor modifications, which would be maintainable for me.

But you are right that it will not support other distributions/RHEL family because it is out of scope for me. It has other limitations, it only supports the APT/.deb installation method, not all netdata settings are configurable through it... and that is by design. Config management is about standardizing environments and having too many config toggles defeats this purpose (and as the number of configuration variables increases, so does the number of non-working configurations).

If you see another use case for which it could be adapted, please let me know (for example I recently added the possibility to configure streaming in a simplified way).

> Ansible is not the only automation tool

This repo could point to other deployment methods/examples/howtos... I think the description/structure should be clarified in this way.

> I would prefer to see this role in Ansible Galaxy. And we will, of course, point out its existence in the documentation.

It's there: https://galaxy.ansible.com/nodiscc/xsrv (as a collection):

```bash
$ ansible-galaxy collection install nodiscc.xsrv
```

But I suggest installing from git instead:

```yaml
# $ cat requirements.yml
collections:
  - name: https://gitlab.com/nodiscc/xsrv.git # from git
    type: git
    version: release
# $ ansible-galaxy collection install --force -r requirements.yml
```

**Pull request merged**
